### PR TITLE
Event history fixes

### PR DESCRIPTION
**Pull request by rymorale**
## Description
- fix searchType logic within the HistoricalEventsQuerying. if we have search type == any, we don't want to do an ordered search.
- fix fnv1a hash creation within MapExtensions. we want to check for empty map values (null or empty string) and not include them when creating the hash.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/core/src/main/java/com/adobe/marketing/mobile/internal/util/MapExtensions.kt [hunk 1]
```diff
@@ -28,8 +28,10 @@ internal fun Map<String, Any?>.fnv1a32(masks: Array<String>? = null): Long {
     if (innerMasks?.isEmpty() == true) innerMasks = null
     innerMasks?.let {
         it.sortedArray().forEach { mask ->
-            if (mask.isNotEmpty() && flattenedMap.containsKey(mask)) {
-                kvPairs.append(mask).append(":").append(flattenedMap[mask].toString())
+            val mapValue = flattenedMap[mask].toString()
+            // we want to ignore null or empty values within the map
+            if (mask.isNotEmpty() && mapValue != "null" && mapValue != "") {
```
Comment (praveek) [inline code/core/src/main/java/com/adobe/marketing/mobile/internal/util/MapExtensions.kt:33]: Can you check for null before converting it to String and add some unit test case to cover this case?

File: code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt [hunk 2]
```diff
@@ -32,7 +32,7 @@ internal fun historicalEventsQuerying(
         var eventCounts = 0
         extensionApi.getHistoricalEvents(
             requests.toTypedArray(),
-            searchType == SEARCH_TYPE_ANY
+            searchType != SEARCH_TYPE_ANY
```
Comment (praveek) [inline code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt:35]: nit: To future proof, can you define a new constant and change it to searchType == SEARCH_TYPE_ORDERED.

Comment (praveek) [reply code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt:35]: Can you also add a test to test this case?

Comment (rymorale) [reply code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt:35]: will get these updated

File: code/core/src/test/java/com/adobe/marketing/mobile/internal/util/MapUtilsTests.java [hunk 3]
```diff
@@ -287,6 +287,92 @@ public void testGetFnv1aHash_WithMaskMatchingNoKeys() {
...
+        // test
+        final long hash =
+                MapUtilsKt.convertMapToFnv1aHash(map, new String[] {"inner.a", "inner.b"});
+        // verify flattened map string "inner.a:1inner.b:2"
+        final long expectedHash = 3328417429L;
+        assertEquals(expectedHash, hash);
+    }
+
+    @Test
+    public void testGetFnv1aHash_NullAndEmptyMapValuesPresent_WithNoMask() {
```
Comment (sbenedicadb) [inline code/core/src/test/java/com/adobe/marketing/mobile/internal/util/MapUtilsTests.java:356]: looks good.  given the newly introduced method acts recursively, let's make sure we add a test which covers the recursive behavior as well

File: code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt [hunk 4]
```diff
@@ -32,7 +32,7 @@ internal fun historicalEventsQuerying(
         var eventCounts = 0
         extensionApi.getHistoricalEvents(
             requests.toTypedArray(),
-            searchType == SEARCH_TYPE_ANY
+            searchType == SEARCH_TYPE_ORDERED
```
Comment (sbenedicadb) [inline code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt:35]: why are we changing the default search type?

**Review by sbenedicadb — CHANGES_REQUESTED** [review]

File: code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt (hunk 4 above)
Comment (sbenedicadb) [reply code/core/src/main/java/com/adobe/marketing/mobile/launch/rulesengine/HistoricalEventsQuerying.kt:35]: ah, nevermind.  this is a bug and you fixed it.  good catch.

File: code/core/src/main/java/com/adobe/marketing/mobile/internal/util/MapExtensions.kt [hunk 5]
```diff
@@ -22,13 +22,14 @@ import org.json.JSONObject
...
-    val flattenedMap = this.flattening()
+    // remove null or empty values before flattening
+    val flattenedMap = this.filterNullOrEmptyValues().flattening()
     val kvPairs = StringBuilder()
     var innerMasks = masks
     if (innerMasks?.isEmpty() == true) innerMasks = null
     innerMasks?.let {
         it.sortedArray().forEach { mask ->
-            if (mask.isNotEmpty() && flattenedMap.containsKey(mask)) {
+            if (mask.isNotEmpty() && flattenedMap[mask] != null) {
```
Comment (praveek) [inline code/core/src/main/java/com/adobe/marketing/mobile/internal/util/MapExtensions.kt:32]: if (mask.isNotEmpty() && !flattenedMap[mask].isNullOrEmptyString())

File: code/core/src/main/java/com/adobe/marketing/mobile/internal/util/MapExtensions.kt [hunk 6]
```diff
@@ -22,13 +22,14 @@ import org.json.JSONObject
...
                 kvPairs.append(mask).append(":").append(flattenedMap[mask].toString())
             }
         }
```
Comment (praveek) [inline code/core/src/main/java/com/adobe/marketing/mobile/internal/util/MapExtensions.kt:35]:
```run {
        flattenedMap.toSortedMap().forEach { entry ->
            if (!entry.value.isNullOrEmptyString()) {
                kvPairs.append(entry.key).append(":").append(entry.value.toString())
            }
        }
    }
```

**Review by praveek — APPROVED** [review]

**Pull request merged**